apps/portals/src/portal-components/RedirectDialog.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react'
+import {
+  Dialog,
+  DialogContent,
+  Button,
+  Link
+} from '@mui/material'
+import { Typography } from 'synapse-react-client'
+
+export type RedirectDialogProps = {
+  onCancelRedirect: ()=>void
+  redirectUrl?: string
+}
+
+export const redirectInstructionsMap = {
+  'https://sites.google.com/sagebase.org/mc2intranet/home?authuser=0' : 
+  <>
+    <Typography variant="body1" sx={{paddingBottom: '20px'}}>
+      <strong>You are currently being redirected to the Multi-Consortia Coordinating (MC2) Center Intranet.</strong>
+    </Typography>
+    <Typography variant="body1" sx={{paddingBottom: '20px'}}>
+      The MC2 Center Intranet is a protected website containing MC2 Center resources. If you are not a participating member of the MC2 Center community, you will not be able to access the intranet.
+    </Typography>
+    <Typography variant="body1" sx={{paddingBottom: '20px'}}>
+      Feel free to <Link target="_blank" rel="noreferrer" href="https://help.cancercomplexity.synapse.org/doc/contact-us">contact the MC2 Center</Link> to learn more, or if you have any questions.
+    </Typography>
+  </>
+}
+
+export const synapseRedirectInstructions =
+  <Typography variant="body1" sx={{paddingBottom: '20px'}}>
+    <strong>You are being redirected to Synapse to view this data.</strong>
+  </Typography>
+
+const isSynapseURL = (url:string) => {
+  if (!url) return false
+  return new URL(url).hostname.toLowerCase() === 'www.synapse.org'
+}
+
+const RedirectDialog = (props: RedirectDialogProps) => {
+  const [countdownSeconds, setCountdownSeconds] = React.useState<number | undefined>()
+  const {redirectUrl, onCancelRedirect} = props
+  const [redirectInstructions, setRedirectInstructions] = React.useState()
+  
+  React.useEffect(() => {
+    if (redirectUrl && countdownSeconds) {
+      // You would expect that we should redirect when countdownSeconds reaches 0,
+      // but it actually takes about a second to perform the redirect. 
+      // So let's start the process when we get to 1.
+      if (countdownSeconds <= 1) {
+        window.location.assign(redirectUrl!)
+      }
+      setTimeout(() => {
+        if (countdownSeconds) {
+          setCountdownSeconds(countdownSeconds => countdownSeconds! - 1)
+        }
+      }, 1000)
+    }
+  }, [redirectUrl, countdownSeconds])
+
+  React.useEffect(() => {
+    if (redirectUrl) {
+      const isRedirectToSynapse = isSynapseURL(redirectUrl)
+      setRedirectInstructions(isRedirectToSynapse ? synapseRedirectInstructions : 
+        redirectInstructionsMap[redirectUrl])
+    }
+  }, [redirectUrl])
+
+  React.useEffect(() => {
+    if (redirectUrl && !countdownSeconds) {
+      setCountdownSeconds(isSynapseURL(redirectUrl) ? 10 : 30)
+    }
+  }, [countdownSeconds, redirectUrl])
+
+  const onClose = () => {
+    // cancel the redirect
+    onCancelRedirect()
+    // and reset countdown seconds
+    setCountdownSeconds(undefined)
```

Comment:
```suggestion
    setCountdownSeconds(isSynapseURL(redirectUrl) ? 10 : 30)
```

Consider creating a variable to store `isSynapseURL(redirectUrl) ? 10 : 30` instead of duplicating it here and in the suggestion above